Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -237,3 +293,48 @@ pickWaypoints waypoints = do
 safeMod :: Int -> Int -> Int
 _ `safeMod` 0 = 0
 a `safeMod` b = a `mod` b
+
+mkDriverFee ::
+  ( MonadReader r m,
+    MonadGuid m,
+    HasField "driverPaymentCycleBuffer" r NominalDiffTime,
+    HasField "driverPaymentCycleDuration" r NominalDiffTime,
+    HasField "driverPaymentCycleStartTime" r NominalDiffTime
+  ) =>
+  UTCTime ->
+  Id DP.Driver ->
+  Money ->
+  m DF.DriverFee
+mkDriverFee now driverId totalAmount = do
+  id <- generateGUID
+  cycleStartCfg <- asks (.driverPaymentCycleStartTime)
+  cycleDurationCfg <- asks (.driverPaymentCycleDuration)
+  bufferDurationCfg <- asks (.driverPaymentCycleBuffer)
+  let startTime = addUTCTime cycleStartCfg (UTCTime (utctDay now) (secondsToDiffTime 0))
+      endTime = addUTCTime cycleDurationCfg startTime
+  return $
+    DF.DriverFee
+      { payBy = addUTCTime bufferDurationCfg endTime,
+        status = DF.ONGOING,
+        createdAt = now,
+        updatedAt = now,
+        ..
+      }
+
+mkPendingPaymentProcessingKey :: UTCTime -> Text
+mkPendingPaymentProcessingKey timestamp = "DriverPendingPaymentProcessing:Timestamp:" <> show timestamp
+
+getPendingPaymentCache :: CacheFlow m r => UTCTime -> m (Maybe Bool)
+getPendingPaymentCache endTime = Hedis.get (mkPendingPaymentProcessingKey endTime)
+
+setPendingPaymentCache :: CacheFlow m r => UTCTime -> m ()
+setPendingPaymentCache endTime = Hedis.setExp (mkPendingPaymentProcessingKey endTime) False 86400
```

Comment:
should this be hardcoded ?

Reply:
It's cache exp time only, it is like this for most of redis keys